### PR TITLE
[BUG] Fixes for sklearn 1.6 and LogisticRegressionClassifier

### PR DESCRIPTION
**Pull request by naoise-h**
This PR provides a number of fixes to maintain compatibility with Scikit-learn 1.6, and also fixing a bug recently reported in `LogisticRegressionClassifier`.

# `LogisticRegressionClassifier`
Prior to Scikit-learn 1.4.0, the loss function for Logistic regression was as follows:
$L = \sum_{i=1}^n \ell(f(x_i), y_i) + \frac{1}{2} \alpha \lVert f \rVert^2.$

The formulation is now as follows:
$L = \frac{1}{n} \sum_{i=1}^n \ell(f(x_i), y_i) + \frac{1}{2n} \alpha \lVert f \rVert^2,$
which gives the following formulation for a DP approximation (from [Chaudhuri et al.](https://www.jmlr.org/papers/volume12/chaudhuri11a/chaudhuri11a.pdf)):
$L_{DP} = \frac{1}{n} \sum_{i=1}^n \ell(f(x_i), y_i) + \frac{1}{2} \Lambda \lVert f \rVert^2 + \frac{1}{n} b^T f + \frac{1}{2}\Delta \lVert f \rVert^2,$
where $\Lambda = \frac{\alpha}{n}$ in Algorithm 2 of the paper. Prior to this PR, the third and fourth terms of $L_{DP}$ were incorrectly scaled, resulting in more noise addition than required for a given $\epsilon$.

As with the previous version, support for non-unit norm of the input data ($x$) is given by appropriately scaling $\epsilon_p^\prime$, and $b$ in Theorem 9.

This change requires Scikit-Learn 1.4 or above and Python 3.9 or above. This fixes two issues, #71 and #97.

### CODE REVIEW

File: .github/workflows/general.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+        python-version: ['3.9', '3.10', '3.11', '3.12']
```
Comment (stefano81) [inline .github/workflows/general.yml]: Suggested to add `3.13` as current main

**Review by stefano81 — APPROVED** [review]
LGTM, just suggested to add support for python 3.13 in a subsequent release.

File: .github/workflows/general.yml (hunk 1 above)
Comment (stefano81) [reply .github/workflows/general.yml]: Marking as resolved as it has been addressed.

Comment (naoise-h) [reply .github/workflows/general.yml]: Done
